### PR TITLE
Fix snap versioning

### LINKED ISSUE #3: Manual version number
In the current version of the snap, the version is set manually and oddly used as a git branch/tag:
https://github.com/openstack-charmers/snap-node-cert-exporter/blob/3bae7e5d9491482a2984957c0fbc92059bb5de8c/snap/snapcraft.yaml#L27

I would expect this to be reversed, so that you specify the git tag and then use a programmatic approach to determine the version number for the snap.

Something like:
```yaml
adopt-info: node-cert-exporter

parts:
  node-cert-exporter:
    source-branch: the-branch-I-want-to-track
    override-stage: |
      craftctl set version ...
      craftctl default
```

There is an example of this in use here:
https://github.com/canonical/snap-ovn-chassis/blob/32d75e4a1ad7d23df50dd04fe4221f419f64db5f/snap/snapcraft.yaml#L226-L228

### PR DESCRIPTION
**Pull request by chanchiwai-ray**
Reverse the process of setting of snap version. The version of the snap should be based on the exporter's git tag, not the other way around. Reference [0]

[0] https://snapcraft.io/docs/using-craftctl

Closes: #3

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -22,10 +22,13 @@ parts:
...
     plugin: go
     source: https://github.com/amimof/node-cert-exporter.git
-    source-tag: $SNAPCRAFT_PROJECT_VERSION
+    source-tag: v1.1.4
     source-type: git
     build-snaps:
       - go
+    override-pull: |
+      craftctl default
+      craftctl set version="$(git describe --tags --abbrev=10)"
```
Comment (fnordahl) [inline snap/snapcraft.yaml]: We most likely do not want the `v` prefix in the upstream tag to be present in the snap version presented to the end user, would there be any way to filter that out from the version string?

**Review by fnordahl — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 1 above)
Comment (chanchiwai-ray) [reply snap/snapcraft.yaml]: Yes, I've included a fix for this.

**Comment by chanchiwai-ray** [comment]
This PR depends on #6 to fix the missing plug issue.

**Review by fnordahl — APPROVED** [review]
LGTM, thanks!

Note that the failing CLA check is just because of making use of GitHub web UI merge master master feature.

**Pull request merged**